newIDE/app/src/CommandPalette/CommandManager.js

```diff
@@ -34,13 +34,19 @@ export type NamedCommandWithOptions<T> = {|
   ...CommandWithOptions<T>,
 |};
 
-export default class CommandManager {
+export interface CommandManagerInterface {
+  commands: { [string]: Command };
+
+  registerCommand: (commandName: string, command: Command) => void;
+  deregisterCommand: (commandName: string) => void;
+  getAllNamedCommands: () => Array<NamedCommand>;
```

Comment:
I added this to avoid a Flow error in the palette UI code, which tries to use `commandManager.getAllNamedCommands()` - which would work because we know its going to be the central manager, but Flow doesn't know that. 

Another option would be to use something like `commandManager.getAllNamedCommands ? commandManager.getAllNamedCommands() : []`. Should I go for that instead?

Reply:
I would rather avoid the ternary and have it part of the interface, especially if it's possible to offer an implementation for `ScopedCommandManager`.